src/Neo/Network/P2P/Payloads/Conditions/CalledByEntryCondition.cs

```diff
@@ -24,7 +24,8 @@ protected override void DeserializeWithoutType(ref MemoryReader reader, int maxN
 
         public override bool Match(ApplicationEngine engine)
         {
-            return engine.CallingScriptHash is null || engine.CallingScriptHash == engine.EntryScriptHash;
+            var state = engine.CurrentContext.GetState<ExecutionContextState>();
+            return state.CallingContext is null || state.CallingContext == engine.EntryContext;
```

Comment:
`ExecuteCall` will create a new context, so it didn't match

Reply:
What do you mean?

Reply:
`CallingContext` is stored in shared state.